openshift-kubecost-install.md

```diff
@@ -0,0 +1,513 @@
+Install Kubecost on Redhat Openshift cluster
+==================
+
+## Overview
+
+This documentation provides instructions on how to deploy Kubecost on Redhat Openshift cluster. Additional security settings may also be required. You can refer to the Architecture overview and installation section for more information. 
+
+---
+## Architecture overview:
+
+Currently there are 2 main options to deploy Kubecost on Redhat Openshift Cluster (OCP).
+
+### Standard deployment:
+
+Kubecost is installed with Cost Analyzer and Prometheus as a time-series database. Data is gathered by the Prometheus installed with Kubecost (user-defined Prometheus). Other metrics are scraped by user-defined Prometheus from OCP monitoring stack managed components like KSM, OSM, CAdvisor, etc .... Kubecost then push and query metrics to/from user-defined Prometheus. Enterprise set up could also work with Thanos as additional component.
+
+The standard deployment is illustrated in the following diagram:
+
+![Standard deployment](images/standard-lightbg.png)
+
+### Grafana managed Prometheus deployment:
+
+Kubecost is installed with the core components only (cost model, frontend) without user-defined Prometheus and other components. Grafana agent is installed as part of the solution to scrape the metrics from OCP monitoring stack managed components and Kubecost `/metrics` endpoint to write the data back to the Grafana Cloud managed Prometheus (Grafana Prometheus) instance. Kubecost reads the metrics directly from Grafana managed Prometheus.
+
+The Grafana managed Prometheus deployment is illustrated in the following diagram:
+
+![Grafana managed Prometheus deployment](images/grafana-agent-lightbg.png)
+
+## Installation instructions:
+
+### Standard deployment:
+
+**1. Clone this repository to your dev environment.**
+
+`git clone https://github.com/kubecost/openshift-helm-chart.git`
+`cd openshift-helm-chart`
+
+**2 Update configuration:**
+
+Replace `CLUSTER_NAME` by your desired value in this file [values-openshift.yaml](https://github.com/kubecost/openshift-helm-chart/blob/main/values-openshift.yaml)
+
+**3. Install Kubecost**
+
+Then install against the local cost-analyzer repo using following helm install command:
+
+```bash
+helm upgrade --install kubecost ./cost-analyzer --namespace kubecost --create-namespace -f ./values-openshift.yaml
+```
+
+Wait for all pods to be ready.
+
+Create a route to the service `kubecost-cost-analyzer` on port `9090` of the `kubecost` project. You can learn more about how to do it on your Openshift portal in this [LINK](https://docs.openshift.com/container-platform/3.11/dev_guide/routes.html#:~:text=to%20the%20router.-,Creating%20Routes,Applications%20section%20of%20the%20navigation.&text=The%20new%20route%20inherits%20the,using%20the%20%2D%2Dname%20option.)
```

Comment:
Would it be straightforward to provide a yaml sample or CLI command to create the route?